slcmd and integration test

zhanggbj opened the pull request:
Hi,

sl --packages and --package-options command need code change in same files, and at the same time, they belong to different stories,  so I submitted two commits in one PR. Thanks for your review:)

[#115478065] {https://www.pivotaltracker.com/story/show/115478065}
[#115478105] {https://www.pivotaltracker.com/story/show/115478105}

Commit:
Added sl --packages command and integration test

[#115478065] (https://www.pivotaltracker.com/story/show/115478065)

Commit:
Address PR #42 comments

[#115478065] (https://www.pivotaltracker.com/story/show/115478065)

Commit:
Added sl --package-options command and integration test

[#115478105]

maximilien commented on cmds/bmp/sl_command.go:
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
 func (cmd slCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+
+	if !cmd.options.Packages && cmd.options.PackageOptions == "" {
```
Hmm, why is cmd.options.Packages now a string?

So when passed you will list all packages?

zhanggbj replied:
I changed cmd.options.Packages from string to bool, so when execute "bmp sl --packages", all packages will be listed.

maximilien replied:
OK cool. I'll assume that's the behavior Jimmy et al. are looking for.

maximilien commented on cmds/bmp/sl_command.go:
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
+	}
+
+	if len(slPackageOptionsResponse.Data.Datacenter) > 0 {
+		cmd.ui.Printf("Packege %s is available in below datacenters:", packageOptions)
```
Typo: "Packege" should be "Package"

zhanggbj replied:
Sorry for my carelessness. Done.

maximilien commented on cmds/bmp/sl_command.go:
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
+		cmd.ui.Printf("Category Code: %s, Name: %s, Required: %t", category.Code, category.Name, category.Required)
+		cmd.ui.Println("")
+
+		table := tablewriter.NewWriter(os.Stdout)
```
See previous issue I mentioned. I don't think we should be passing os.Stdout. It should come from cmd.Ui. Unless you open a chore to fix this later you need to fix this with the others. Thx.

zhanggbj replied:
I opened a chore #119336713 to fix this, as the code change about moving this to cmd.ui is in PR #38, so I didn't fix it here.

maximilien commented on cmds/bmp/sl_command_test.go:
```diff
@@ -65,18 +66,183 @@ var _ = Describe("sl command", func() {
+					}
+				})
+
+				It("validation pass without errors", func() {
```
This needs in correct English, so, this is better:

'''
It("passes validation without errors", ...)
'''

zhanggbj replied:
Done.Thanks.

maximilien commented on cmds/bmp/sl_command_test.go:
```diff
@@ -65,18 +66,183 @@ var _ = Describe("sl command", func() {
+					}
+				})
+
+				It("validation fails with errors", func() {
```
It("fails validation with errors", ...)

zhanggbj replied:
Done

Commit:
Updated test case description

[#115478065] {https://www.pivotaltracker.com/story/show/115478065}
[#115478105] {https://www.pivotaltracker.com/story/show/115478105}

Commit:
Merge remote-tracking branch 'upstream/master' into WIP-slcmd

maximilien commented on cmds/bmp/sl_command.go:
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
+		return slPackagesResponse.Status, nil
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
```
You need to switch this to using the `cmd.UI`. I'll accept but please change.
